tidb-cloud/migrate-sql-shards.md

```diff
@@ -0,0 +1,596 @@
+
+```sql
+mysql> CREATE TABLE `sales` (
+   ->   `id` bigint(20) NOT NULL ,
+   ->   `uid` varchar(40) NOT NULL,
+   ->   `sale_num` bigint DEFAULT NULL,
+   ->   INDEX (`id`),
+   ->   UNIQUE KEY `ind_uid` (`uid`)
+   -> );
+Query OK, 0 rows affected (0.17 sec)
+```
+
+For more information about the solutions to solve such conflicts, see [Remove the PRIMARY KEY attribute from the column](https://docs.pingcap.com/tidb/stable/shard-merge-best-practices#remove-the-primary-key-attribute-from-the-column).
+
+### Step 4. Configure Amazon S3 access
+
+To configure Amazon S3 access, follow the instructions in [Configure Amazon S3 access](/tidb-cloud/config-s3-and-gcs-access.md#configure-amazon-s3-access).
+
+The following example only lists key policy configurations. Replace the Amazon S3 path with your own values.
+
+```yaml
+{
+   "Version": "2012-10-17",
+   "Statement": [
+       {
+           "Sid": "VisualEditor0",
+           "Effect": "Allow",
+           "Action": [
+               "s3:GetObject",
+               "s3:GetObjectVersion"
+           ],
+           "Resource": [
+               "arn:aws:s3:::dumpling-s3/*"
+           ]
+       },
+       {
+           "Sid": "VisualEditor1",
+           "Effect": "Allow",
+           "Action": [
+               "s3:ListBucket",
+               "s3:GetBucketLocation"
+           ],
+
+           "Resource": "arn:aws:s3:::dumpling-s3"
+       }
+   ]
+}
+```
+
+### Step 5. Perform the data import task
+
+After configuring the Amazon S3 access, you can perform the data import task in the [TiDB Cloud console](https://tidbcloud.com/console/clusters).
+
+1. Navigate to the **Active Clusters** page for your project.
+
+2. In the TiDB Cloud Admin console, choose a target project and a target cluster deployed on the Google Cloud Platform. Then click **...** in the upper-right corner of the cluster area and select **Import Data**.
+
+3. On the **Data Import** page, fill in the following information:
+
+    - **Data Format**: select **CSV**.
+    - **Location**: `AWS`
+    - **Bucket URL**: fill in the bucket URL of your source data. You can use the second-level directory corresponding to tables, `s3://dumpling-s3/store/sales` in this example, so that TiDB Cloud can import and merge the data in all MySQL instances into `store.sales` in one go.
+    - **Role-ARN**: enter the Role-ARN you obtained.
+    - **Target Cluster**: shows the cluster name and the region name.
+
+    If the location of the bucket is different from your cluster, confirm the compliance of cross region. Click **Next**.
+
+    TiDB Cloud starts validating whether it can access your data in the specified bucket URL. If you get the `AccessDenied` error, see [Troubleshoot Access Denied Errors during Data Import from S3](/tidb-cloud/troubleshoot-import-access-denied-error.md).
```

Comment:
```suggestion
    TiDB Cloud starts validating whether it can access your data in the specified bucket URL. After validation, TiDB Cloud tries to scan all the files in the data source using the default file naming pattern, and returns a scan summary result on the left side of the next page. If you get the `AccessDenied` error, see [Troubleshoot Access Denied Errors during Data Import from S3](/tidb-cloud/troubleshoot-import-access-denied-error.md).
```